src/firewall/core/fw_ifcfg.py

```diff
@@ -36,11 +36,17 @@ def search_ifcfg_of_interface(interface):
     if not os.path.exists(config.IFCFGDIR):
         return None
 
+    # RedHat puts ifcfg files in 'network-scripts'
+    suseifcfg = not "network-scripts" in config.IFCFGDIR
     filename = "%s/ifcfg-%s" % (config.IFCFGDIR, interface)
     if os.path.exists(filename):
         ifcfg_file = ifcfg(filename)
         ifcfg_file.read()
-        if ifcfg_file.get("DEVICE") == interface:
+        # on SUSE, ETHERDEVICE is only applicable when creating VLANs so it may not always
+        # be there.
+        if ifcfg_file.get("DEVICE") == interface or \
+           ifcfg_file.get("ETHERDEVICE") == interface or \
```

Comment:
For SUSE ETHERDEVICE will be the underlying interface. Not the name of the VLAN interfaces, so this won't match. i.e. file ifcfg-vlan100 may contain ETHERDEVICE=eth0